common/static/common/js/spec/components/paging_collection_spec.js

```diff
@@ -0,0 +1,238 @@
+        'common/js/spec_helpers/ajax_helpers',
+        'common/js/spec_helpers/spec_helpers'
+    ],
+    function ($, Backbone, _, URI, PagingCollection, AjaxHelpers, SpecHelpers) {
+        'use strict';
+
+        describe('PagingCollection', function () {
+            var collection,
+                requests,
+                server = {
+                    isZeroIndexed: false,
+                    count: 43,
+                    respond: function () {
+                        var params = (new URI(requests[requests.length - 1].url)).query(true),
+                            page = parseInt(params['page'], 10),
+                            page_size = parseInt(params['page_size'], 10),
+                            page_count = Math.ceil(this.count / page_size);
+
+                        // Make zeroPage consistently start at zero for ease of calculation
+                        var zeroPage = page - (this.isZeroIndexed ? 0 : 1);
+                        if (zeroPage < 0 || zeroPage > page_count) {
+                            AjaxHelpers.respondWithError(requests, 404, {}, requests.length - 1);
+                        } else {
+                            AjaxHelpers.respondWithJson(requests, {
+                                'count': this.count,
+                                'current_page': page,
+                                'num_pages': page_count,
+                                'start': zeroPage * page_size,
+                                'results': []
+                            }, requests.length - 1);
+                        }
+                    }
+                },
+                assertQueryParams = function (params) {
+                    var urlParams = (new URI(requests[requests.length - 1].url)).query(true);
+                    _.each(params, function (value, key) {
+                        expect(urlParams[key]).toBe(value);
+                    });
+                };
+
+            beforeEach(function () {
+                collection = new PagingCollection();
+                collection.perPage = 10;
+                requests = AjaxHelpers.requests(this);
+                server.isZeroIndexed = false;
+                server.count = 43;
+            });
+
+            it('can register sortable fields', function () {
+                collection.registerSortableField('test_field', 'Test Field');
+                expect('test_field' in collection.sortableFields).toBe(true);
+                expect(collection.sortableFields['test_field'].displayName).toBe('Test Field');
+            });
+
+            it('can register filterable fields', function () {
+                collection.registerFilterableField('test_field', 'Test Field');
+                expect('test_field' in collection.filterableFields).toBe(true);
+                expect(collection.filterableFields['test_field'].displayName).toBe('Test Field');
+            });
+
+            it('can set the sort field and get the display name', function () {
```

Comment:
Nit: I'd drop the ``and get the display name`` part of this name. That is just one aspect that should be correct after setting the sort field.